core/arch/arm/tee/entry_std.c

```diff
@@ -63,8 +63,11 @@ static TEE_Result set_tmem_param(const struct optee_msg_param_tmem *tmem,
 	paddr_t pa = READ_ONCE(tmem->buf_ptr);
 	size_t sz = READ_ONCE(tmem->size);
 
-	/* Handle NULL memory reference */
-	if (!pa && !sz) {
+	/*
+	 * Handle NULL memory reference
+	 * NOTE: sz==0 is a valid case, must pass valid ptr to TA
```

Comment:
ok sz==0 be a valid case, but I don't think we **must pass** a valid pointer.
This only interest for a client to pass a null sized buffer to a TA, is to get from the TA the expected size of an output buffer. In this case, the TA does not need to access the buffer (get it mapped), only to output the requested buffer byte size. Your thoughts?

Reply:
Not sure I follow you fully here @etienne-lms ? Even in the case where you poll for a size, you might have checks for NULL etc on the TA side. We do not need to access any buffer, but the buffer from TA point of view must at least be a NULL buffer.

Reply:
ok @jbech-linaro.

A minor comment on the inline comment: can remove the sentence `NOTE: sz==0 is a valid case, must pass valid ptr to TA`. I think it describes this change but is not relevant at this place in the source file.

Reply:
ok

Reply:
Passing an invalid pointer is a bit harder. Do you think it's important that it's invalid?